### DISCUSSION TITLE
Advice: Oddly shaped items

### DISCUSSION THREAD

**Discussion by tipx2** [peter-kish/gloot#180] [Q&A] [answered]
Hi, this is not explicitly supported by Gloot but I was wonder what might be the best way to go about creating more oddly shaped items for a grid inventory within Gloot for a game similar to "Backpack Hero" or "Backpack Battles". For example, the shapes below:
![image](https://github.com/peter-kish/gloot/assets/57505742/8b11eef7-b418-4119-8e07-2a46b6e5eae8)
I had a brief look at the code involving the shape of the item but it looks like that is quite set in stone to be a Vector2i. However perhaps there is a good way of joining 2 shapes together and treating them as one? (allowing them to rotate together etc.) but that might be a bit complex. Would just like to know if there is a simple/easy way to do this. Thanks.

**Comment by peter-kish** [comment, marked as answer]
There is definitely no easy way to do this with the way things are currently implemented. And unfortunately I don't see something like this being implemented any time soon :(
The main goal of this plugin is to help developers make generic inventories, while this is quite an exotic feature.

**Reply by tipx2** [reply]
Ok, that's totally fair! I will look about making my own simple system for this :)
